recipe-client-addon/lib/PreferenceExperiments.jsm

```diff
@@ -142,6 +142,16 @@ this.PreferenceExperiments = {
       previousPreferenceValue: Preferences.get(preferenceName, undefined),
     };
 
+    if (experiment.previousPreferenceValue) {
+      const previousType = typeof experiment.previousPreferenceValue;
+      const givenType = typeof preferenceValue;
```

Comment:
> @Osmose It doesn't seem that I can just give the method a value and have it spit the type of out; it takes a pref name and checks the current value of that pref.

Are you sure? I just ran this in my browser console and it worked fine:

```js
>> Services.prefs.getPrefType("asdf") // Boolean pref
128
>> Services.prefs.getPrefType("asdf") === Services.prefs.PREF_BOOL
true
>> Services.prefs.getPrefType("asdfasdf") // Doesn't exist
0
>> Services.prefs.getPrefType("asdfasdf") === Services.prefs.PREF_INVALID
true
```

Note that it's returning ints. The doc page mentioned that the return values were `PREF_BOOL` and the like, which I found as constants on the `Services.prefs` object. Sorry for not checking and explaining that earlier.